macros/src/main/scala/com/gu/contentapi/circe/CirceScroogeMacros.scala

```diff
@@ -187,4 +281,49 @@ object CirceScroogeMacros {
     tree
 
   }
+
+  implicit def encodeThriftUnion[A <: ThriftUnion]: Encoder[A] = macro encodeThriftUnion_impl[A]
+
+  def encodeThriftUnion_impl[A: c.WeakTypeTag](c: blackbox.Context): c.Tree = {
+    import c.universe._
+
+    val A = weakTypeOf[A]
+
+    val memberClasses: Iterable[Symbol] = A.companion.members.filter { member =>
```

Comment:
Thanks for taking a look @philwills !
Looks like using macro bundles will definitely tidy things up. I think I'll do that work in a separate PR though.